src/main/java/io/stargate/sgv2/jsonapi/service/projection/ProjectionLayer.java

```diff
@@ -195,21 +189,22 @@ void addSlicer(boolean failOnOverlap, String fullPath, String segment, Slicer sl
    * @return {@code true} if path is included; {@code false} if not.
    */
   public boolean isPathIncluded(String path) {
-    final String[] segments = DOT.split(path);
+    ProjectionPath p = ProjectionPath.from(path);
+    final List<String> segments = p.getSegments();
     return isPathIncluded(segments, 0);
   }
 
-  private boolean isPathIncluded(String[] segments, int index) {
+  private boolean isPathIncluded(List<String> segments, int index) {
```

Comment:
... and if we have `ProjectionPath.segment()` can (and should) just pass `ProjectionPath` here.